### ISSUE TITLE
GeoJsonFeatureCollection.toJson does not return original JSON from the response

### ISSUE THREAD

**Issue by miklcct** [Dhi13man/open_route_service#19]
I need to use another GeoJSON library to parse the response of `openRouteService.geocodeAutoCompleteGet()`, so I call `FeatureCollection.fromJson(predictions.toJson())`. where `FeatureCollection` is from another library.

However, the `toJson()` generated from this library is invalid GeoJSON, so it can't be parsed.

**Comment by Dhi13man** [comment]
If possible, can you provide me a sample of what kind of GeoJSON is required and how it is different form the JSON generated by this library? We can work on making it work.

**Comment by miklcct** [comment]
Test file:
```dart
import 'package:geodart/geometries.dart';
import 'package:open_route_service/open_route_service.dart';

void main() {
  final original = Point(Coordinate(51.5, 0.0));
  final validJson = original.toJson();
  print(validJson);
  final imported = GeoJsonFeature.fromJson(validJson);
  final exported = imported.toJson();
  print(exported);
  final backToOriginal = Point.fromJson(exported);
  print(backToOriginal.toJson());
}
```

Output:
```
{type: Feature, geometry: {type: Point, coordinates: [0.0, 51.5]}, properties: {}}
{type: Feature, properties: {}, geometry: {type: Point, coordinates: [[[0.0, 51.5, 0.0]]]}}
Unhandled exception:
RangeError (index): Invalid value: Only valid value is 0: 1
#0      List.[] (dart:core-patch/growable_array.dart:264:36)
#1      new Point.fromJson (package:geodart/src/featureTypes/point.dart:81:49)
#2      main (package:app/test.dart:11:32)
#3      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:297:19)
#4      _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)

Process finished with exit code 255
```

As seen, the second line of the output is invalid GeoJSON, because `coordinates` of a `Point` must be a list of 2 numbers.

**Comment by Dhi13man** [comment]
Hey @miklcct. #20 should fix this. Also refactored the code to make it slightly more readable. Would love it if you gave it a review and tested it out. Will merge it and get it published in the next 1-2 days.

Regardless, huge thanks for raising the issue and contributing the test case.

**Comment by Dhi13man** [comment]
This fix has been published as version 1.2.5. Thanks again
